src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -837,7 +842,12 @@ private void maybeFail(ResultSetBuilder result, QueryOptions options)
             String clientMsg = msg + " with " + loggableTokens(options);
             ClientWarn.instance.warn(clientMsg);
             logger.warn("{} with query {}", msg, asCQL(options));
-            cfs().metric.clientReadSizeAborts.mark();
+            ColumnFamilyStore store = cfs();
+            if (store != null)
+            {
+                store.metric.clientReadSizeAborts.mark();
+                store.metric.clientReadSize.update(result.getSize());
```

Comment:
We're guaranteed not to update `clientReadSize` twice because we'll throw an exception and skip `maybeWarn()`, correct?

Reply:
we warn once and have abort checks through the logic; so will not double log (and this is tested =D).

so yes, since the exception blocks us from calling warn, we don't double count